### PR TITLE
feat: introduce `go-parser` service

### PR DESCRIPTION
**Pull request by wrn14897**
Introduce a new separated go-parser service that obfuscates `db.statement` with sql lexer

### CODE REVIEW

File: docker/ingestor/core.toml [hunk 1]
```diff
@@ -706,6 +706,20 @@ if is_object(.r) {
...
+condition = '''
+  ("${ENABLE_GO_PARSER:-false}" == "true" && is_nullish(.b."db.statement")) || "${ENABLE_GO_PARSER:-false}" == "false"
+'''
+
+[transforms.go_spans]
+type = "filter"
+inputs = ["spans"]
+condition = '''
+  "${ENABLE_GO_PARSER:-false}" == "true" && !is_nullish(.b."db.statement")
+'''
```
Comment (wrn14897) [inline docker/ingestor/core.toml:722]: filters are controlled by `ENABLE_GO_PARSER` flag and currently its not enabled

File: packages/go-parser/main.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+					continue
+				}
+				normalized, _, err := normalizer.Normalize(dbStatement.(string))
+				if err != nil {
+					fmt.Println("Error normalizing SQL:", err)
+					continue
+				}
+				obfuscator := sqllexer.NewObfuscator()
+				obfuscated := obfuscator.Obfuscate(normalized)
+				log["b"].(map[string]interface{})["db.sql.normalized"] = obfuscated
```
Comment (wrn14897) [inline packages/go-parser/main.go:104]: Want to double check this is the custom property name

File: docker/ingestor/http-sinks.toml [hunk 3]
```diff
@@ -1,7 +1,18 @@
+[sinks.go_parser]
+type = "http"
+uri = "http://go-parser:7777"
+inputs = ["go_spans"] # only send spans for now
+compression = "gzip"
+encoding.codec = "json"
+batch.max_bytes = 10485760 # 10MB, required for rrweb payloads
```
Comment (jaggederest) [inline docker/ingestor/http-sinks.toml:7]: Do we want rrweb spans? They should never have a db.statement right? Just double checking

File: packages/go-parser/main.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+var (
+	VERSION        = "0.0.1"
+	PORT           = os.Getenv("PORT")
+	AGGREGATOR_URL = os.Getenv("AGGREGATOR_API_URL")
+	// https://opentelemetry.io/docs/specs/semconv/database/database-spans/#:~:text=db.system%20has%20the%20following%20list%20of%20well%2Dknown%20values
+	NON_SQL_DB_SYSTEMS = []string{
+		"adabas",
+		"filemaker",
+		"coldfusion",
+		"cassandra",
```
Comment (jaggederest) [inline packages/go-parser/main.go:29]: I think Cassandra is close enough to parse with sql-lexer, for now.

File: packages/go-parser/main.go [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+		"hbase",
+		"mongodb",
+		"redis",
+		"couchbase",
```
Comment (jaggederest) [inline packages/go-parser/main.go:33]: Couchbase is sql-like

**Review by jaggederest — DISMISSED** [review]
Great stuff!

**Review by MikeShi42 — APPROVED** [review]

File: docker/ingestor/http-sinks.toml (hunk 3 above)
Comment (wrn14897) [reply docker/ingestor/http-sinks.toml:7]: yeah, it was copy-n-paste. no big deal at this moment

**Pull request merged**